### PR TITLE
installation_proxy: implement .ipcc installation

### PR DESCRIPTION
**Pull request by abdullah-albanna**
both ipcc and ipa are tested

### CODE REVIEW

File: pymobiledevice3/cli/apps.py [hunk 1]
```diff
@@ -45,10 +45,10 @@ def uninstall(service_provider: LockdownClient, bundle_id):
 
 
 @apps.command('install', cls=Command)
-@click.argument('ipa_or_app_path', type=click.Path(exists=True))
-def install(service_provider: LockdownServiceProvider, ipa_or_app_path: str) -> None:
-    """ install given .ipa/.app """
-    InstallationProxyService(lockdown=service_provider).install_from_local(ipa_or_app_path)
+@click.argument('ipa_or_app_or_ipcc_path', type=click.Path(exists=True))
```
Comment (doronz88) [inline pymobiledevice3/cli/apps.py]: It's time to simplify this:
```suggestion
@click.argument('package', type=click.Path(exists=True))
```

File: pymobiledevice3/cli/apps.py [hunk 2]
```diff
@@ -45,10 +45,10 @@ def uninstall(service_provider: LockdownClient, bundle_id):
...
+def install(service_provider: LockdownServiceProvider, ipa_or_app_or_ipcc_path: str) -> None:
+    """ install given .ipa/.app or .ipcc """
```
Comment (doronz88) [inline pymobiledevice3/cli/apps.py]:
```suggestion
    """ install given .ipa/.app/.ipcc """
```

File: pymobiledevice3/services/installation_proxy.py [hunk 3]
```diff
@@ -56,6 +58,10 @@ def _watch_completion(self, handler: Callable = None, *args) -> None:
                     handler(completion, *args)
                 self.logger.info(f'{response.get("PercentComplete")}% Complete')
             if response.get('Status') == 'Complete':
+                if ipcc:
+                    # there is no progress when installing a .ipcc file, 
+                    # so we just put a simple message indicating it's done
+                    self.logger.info('Installation succeed. ')
```
Comment (doronz88) [inline pymobiledevice3/services/installation_proxy.py]:
```suggestion
                    self.logger.info('Installation succeed.')
```

File: pymobiledevice3/services/installation_proxy.py [hunk 4]
```diff
@@ -72,9 +78,9 @@ def send_cmd_for_bundle_identifier(self, bundle_identifier: str, cmd: str = 'Arc
         self.service.send_plist(cmd)
         self._watch_completion(handler, *args)
 
-    def install(self, ipa_path: str, options: Optional[dict] = None, handler: Callable = None, *args) -> None:
-        """ install given ipa from device path """
-        self.install_from_local(ipa_path, 'Install', options, handler, args)
+    def install(self, ipa_or_ipcc_path: str, options: Optional[dict] = None, handler: Callable = None, *args) -> None:
```
Comment (doronz88) [inline pymobiledevice3/services/installation_proxy.py]:
```suggestion
    def install(self, package_path: str, options: Optional[dict] = None, handler: Callable = None, *args) -> None:
```

File: pymobiledevice3/services/installation_proxy.py [hunk 5]
```diff
@@ -88,26 +94,62 @@ def uninstall(self, bundle_identifier: str, options: Optional[dict] = None, hand
         """ uninstall given bundle_identifier """
         self.send_cmd_for_bundle_identifier(bundle_identifier, 'Uninstall', options, handler, args)
 
-    @str_to_path('ipa_or_app_path')
-    def install_from_local(self, ipa_or_app_path: Path, cmd: str = 'Install', options: Optional[dict] = None,
+    @str_to_path('ipa_or_app_or_ipcc_path')
+    def install_from_local(self, ipa_or_app_or_ipcc_path: Path, cmd: str = 'Install', options: Optional[dict] = None,
```
Comment (doronz88) [inline pymobiledevice3/services/installation_proxy.py]:
```suggestion
    def install_from_local(self, package_path: Path, cmd: str = 'Install', options: Optional[dict] = None,
```

**Review by doronz88 — CHANGES_REQUESTED** [review]
Thanks for the PR! Please fix the requested suggestions

**Comment by abdullah-albanna** [comment]
Tried to be more explicit. Should be good now

**Review by doronz88 — APPROVED** [review]

**Comment by doronz88** [comment]
Thanks for the contribution! :)

**Pull request merged**